src/lib/spriteActor.js

```diff
@@ -0,0 +1,50 @@
+async function loadImage(src) {
+    let img = new Image();
+    img.src = src;
+    return new Promise((res, rej) => {
+        img.onload = () => {
+            res(img);
+        }
+    });
+}
+
+export default class SpriteActor extends Actor{
+    constructor(bounds, spriteObj){
+        super(bounds);
+        this.properties = spriteObj.properties || {};
+        this.states = spriteObj.states || {idle: {location: null, rps: 5, loop: true, frames: [{x: 0, y: 0, width: 0, height: 0}]}};
+
+    }
+
+    loadSpriteObj = (spriteObj) => {
+        this.properties = spriteObj.properties;
+        this.states = spriteObj.states;
+
+        //what is diffing images?
```

Comment:
Like if there are different or same sprite images:
- if they are the same don't load them
- else load them 

Don't worry about this too much, most spriteObj will be loaded once in init